code/hsec-tools/src/Security/OSV.hs

```diff
@@ -0,0 +1,223 @@
+-- | This module contains the OSV datatype and its ToJSON instance.
+-- The module was initialized with http://json-to-haskell.chrispenner.ca/
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Security.OSV where
+
+import Data.Aeson (ToJSON(..), FromJSON(..), Value(..), (.:), (.=), object)
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Text (Text)
+
+data Affected = Affected
+  { affectedRanges :: [Ranges]
+  , affectedPackage :: Package
+  , affectedEcosystemSpecific :: EcosystemSpecific
+  , affectedDatabaseSpecific :: DatabaseSpecific
+  } deriving (Show, Eq, Ord)
+
+data Affects = Affects
+  { affectsOs :: [Value]
+  , affectsArch :: [Value]
+  , affectsFunctions :: [Value]
+  } deriving (Show, Eq, Ord)
+
+data DatabaseSpecific = DatabaseSpecific
+  { databaseSpecificCvss :: Maybe Value
+  , databaseSpecificCategories :: [Value]
+  , databaseSpecificInformational :: Text
+  } deriving (Show, Eq, Ord)
+
+data EcosystemSpecific = EcosystemSpecific
+  { ecosystemSpecificAffects :: Affects
+  } deriving (Show, Eq, Ord)
+
+data Events = Events
+  { eventsIntroduced :: Text
+  } deriving (Show, Eq, Ord)
+
+data Model = Model
```

Comment:
I think we are missing the `schema_version` and `credits` fields
- https://ossf.github.io/osv-schema/#schema_version-field
- https://ossf.github.io/osv-schema/#credits-fields

Reply:
Also missing `withdrawn` and a general `database_specific` field (that applies to the entire vuln, not to a specific affected version).

https://ossf.github.io/osv-schema/#withdrawn-field
https://ossf.github.io/osv-schema/#database_specific-field

Reply:
I added these in #42 (which is based on this PR).